corehq/apps/users/util.py

```diff
@@ -45,6 +50,40 @@
 ]
 
 
+def generate_mobile_username(username, domain):
+    """
+    Returns the email formatted mobile username if successfully generated
+    Handles exceptions raised by .validation.validate_mobile_username with user facing messages
+    Any additional validation should live in .validation.validate_mobile_username
+    :param username: required str, expecting the first part of a mobile username
+    :param domain: required str, domain name
+    :return: str, email formatted mobile username
+    Example use: generate_mobile_username('username', 'domain') -> 'username@domain.commcarehq.org'
+    """
+    from .validation import validate_mobile_username
+    error = None
+    try:
+        return validate_mobile_username(username, domain)
+    except InvalidUsernameException:
+        error = _("Username '{}' may not contain special characters.").format(username)
+        if '..' in username:
+            error = _("Username '{}' may not contain consecutive '.' (period).").format(username)
+        elif username.endswith('.'):
+            error = _("Username '{}' may not end with a '.' (period).").format(username)
+    except UsernameAlreadyExists as e:
+        if e.is_deleted:
+            error = _("Username '{}' belonged to a user that was deleted and cannot be reused.").format(username)
+        else:
+            error = _("Username '{}' is already taken.").format(username)
+    except ReservedUsernameException:
+        error = _("Username '{}' is reserved.").format(username)
+    except ValueError:
+        error = _("Must specify a username.")
```

Comment:
nit: consistent language in errors
```suggestion
        error = _("Username is required")
```